src/main/java/org/carlmontrobotics/subsystems/Arm.java

```diff
@@ -1,48 +1,263 @@
-import edu.wpi.first.wpilibj2.command.WaitCommand;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
 
+// Arm angle is measured from horizontal on the intake side of the robot and bounded between -3π/2 and π/2
+// Wrist angle is measured relative to the arm with 0 being parallel to the arm and bounded between -π and π (Center of Mass of Roller)
 public class Arm extends SubsystemBase {
-    public Arm() {
-			//arm 
-      /*
-       have 3 set positions
-       Speaker position 
-       Amp position
-        ground position
-       These set positions would also be conrtolled by buttons
-
-       There will also be a manual control for the arm using the right joystick
-      */
-    }
-	
-    public void setArmPos() {
-      //Sets arm to the optimal angle for amp, speaker and Ground intake | used to score in amp
-      //these values are in constants
-      //pass in where scorign and use switch statement
+
+    private final CANSparkMax armMotorMaster/*left */ = MotorControllerFactory.createSparkMax(ARM_MOTOR_PORT_MASTER, MotorConfig.NEO);
+    private final CANSparkMax armMotorFollower/*right */ = MotorControllerFactory.createSparkMax(ARM_MOTOR_PORT_FOLLOWER, MotorConfig.NEO);
+    private final SparkAbsoluteEncoder armMasterEncoder = armMotorMaster.getAbsoluteEncoder(SparkAbsoluteEncoder.Type.kDutyCycle);
+    private final RelativeEncoder armFollowEncoder = armMotorFollower.getEncoder();
+    private final MutableMeasure<Voltage> voltage = mutable(Volts.of(0));
+    private final MutableMeasure<Velocity<Angle>> velocity = mutable(RotationsPerSecond.of(0));
+    private final MutableMeasure<Angle> distance = mutable(Rotations.of(0));
+
+
     
+    private static double kDt = 0.02;
+   
+    //PID, feedforward, trap profile
+
+    // rel offset = starting absolute offset
+    private final ArmFeedforward armFeed = new ArmFeedforward(kS, kG, kV, kA);
+    private final SparkPIDController armPID1 = armMotorMaster.getPIDController();
+    private final SparkPIDController armPID2 = armMotorFollower.getPIDController();
+
+    private TrapezoidProfile armProfile = new TrapezoidProfile(TRAP_CONSTRAINTS);
+    TrapezoidProfile.State goalState = new TrapezoidProfile.State(0,0);//TODO: update pos later
+
+    // rad, rad/s
+    //public static TrapezoidProfile.State[] goalState = { new TrapezoidProfile.State(-Math.PI / 2, 0), new TrapezoidProfile.State(0, 0) };
+
+    public Arm() {
+      // weird math stuff
+        armMotorMaster.setInverted(MOTOR_INVERTED_MASTER);
+        armMotorMaster.setIdleMode(IdleMode.kBrake);
+        armMotorFollower.setInverted(MOTOR_INVERTED_FOLLOWER);
+        armMotorFollower.setIdleMode(IdleMode.kBrake);
+        
+        armMasterEncoder.setPositionConversionFactor(ROTATION_TO_RAD);
+        armMasterEncoder.setVelocityConversionFactor(ROTATION_TO_RAD);
+        armMasterEncoder.setInverted(ENCODER_INVERTED);
+
+        armMotorFollower.follow(armMotorMaster);
```

Comment:
As a heads up, since you make the follow motor follow master, the pid for the follow motor won't actually do anything

Reply:
i stand by that you should not have PID for the follow motor since they are powering the same thing through a chain. It makes things weird if the motors are given different voltages. @FriedLongJohns thinks that the chain is not a "hard connection" between both motors so they should run at different coltages. @brettle thoughts on this?

Reply:
Possible issues such as just a bad motor or slower wire could affect the arm in having only one pid controller. Possible issue is bad fab as intake had issues with a motor skipping therefore causing it to be slower. Although it doesn't have the equivalent of a belt there could be issues where the slave motor and the master motor have a difference in faulty builds, making two pid controller helpful